plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/params/BuildImageParams.java

```diff
@@ -182,6 +188,9 @@ public BuildImageParams withFiles(@NotNull File... files) {
      *         if {@code files} is null
      */
     public BuildImageParams addFiles(@NotNull File... files) {
+        if (remote != null) {
```

Comment:
No, `addFiles` is public and can be used directly. If we created params from `remote` and try to add a file we'll get `NPE`. It is not clear as for me.